include/fastdds/rtps/transport/shared_mem/SharedMemTransportDescriptor.h

```diff
@@ -24,77 +24,102 @@ namespace rtps {
 class TransportInterface;
 
 /**
- * Shared memory transport configuration
+ * Shared memory transport configuration.
+ * The kind value for SharedMemTransportDescriptor is given by eprosima::fastrtps::rtps::LOCATOR_KIND_SHM.
+ *
+ * - segment_size_: size of the shared memory segment (in octets).
+ *
+ * - port_queue_capacity_: size of the listening port (in messages).
+ *
+ * - healthy_check_timeout_ms_: timeout for the health check of ports (ms).
+ *
+ * - rtps_dump_file_: full path of the protocol dump file.
  *
  * @ingroup TRANSPORT_MODULE
  */
-typedef struct SharedMemTransportDescriptor : public TransportDescriptorInterface
+struct SharedMemTransportDescriptor : public TransportDescriptorInterface
 {
-    virtual ~SharedMemTransportDescriptor() 
-    {
-        
-    }
+    //! Destructor
+    virtual ~SharedMemTransportDescriptor() = default;
 
     virtual TransportInterface* create_transport() const override;
+
+    //! Minimum size of the send buffer
     uint32_t min_send_buffer_size() const override
     {
         return 0;
     }
-    
+
+    //! Constructor
     RTPS_DllAPI SharedMemTransportDescriptor();
 
+    //! Copy constructor
     RTPS_DllAPI SharedMemTransportDescriptor(
-            const SharedMemTransportDescriptor& t);
+            const SharedMemTransportDescriptor& t) = default;
+
+    //! Copy assignment
+    SharedMemTransportDescriptor& operator =(
```

Comment:
```suggestion
    RTPS_DllAPI SharedMemTransportDescriptor& operator =(
```

Reply:
Done in 2bb1c5feceb7be770327e77413667687e841cd7f